openmc/plotter.py

```diff
@@ -0,0 +1,537 @@
+            nucT = "{}K".format(int(round(data_Ts[closest_t])))
+
+        # Prep S(a,b) data if needed
+        if sab_name:
+            sab = openmc.data.ThermalScattering.from_hdf5(sab_name)
+            # Obtain the nearest temperature
+            if strT in sab.temperatures:
+                sabT = strT
+            else:
+                data_Ts = sab.temperatures
+                for t in range(len(data_Ts)):
+                    # Take off the "K" and convert to a float
+                    data_Ts[t] = float(data_Ts[t][:-1])
+                min_delta = np.finfo(np.float64).max
+                closest_t = -1
+                for t in data_Ts:
+                    if abs(data_Ts[t] - T) < min_delta:
+                        closest_t = t
+                sabT = "{}K".format(int(round(data_Ts[closest_t])))
+
+            # Create an energy grid composed the S(a,b) and
+            # the nuclide's grid
+            grid = nuc.energy[nucT]
+            sab_Emax = 0.
+            sab_funcs = []
+            if sab.elastic_xs:
+                elastic = sab.elastic_xs[sabT]
+                if isinstance(elastic, openmc.data.CoherentElastic):
+                    grid = np.union1d(grid, elastic.bragg_edges)
+                    if elastic.bragg_edges[-1] > sab_Emax:
+                        sab_Emax = elastic.bragg_edges[-1]
+                elif isinstance(elastic, openmc.data.Tabulated1D):
+                    grid = np.union1d(grid, elastic.x)
+                    if elastic.x[-1] > sab_Emax:
+                        sab_Emax = elastic.x[-1]
+                sab_funcs.append(elastic)
+            if sab.inelastic_xs:
+                inelastic = sab.inelastic_xs[sabT]
+                grid = np.union1d(grid, inelastic.x)
+                if inelastic.x[-1] > sab_Emax:
+                        sab_Emax = inelastic.x[-1]
+                sab_funcs.append(inelastic)
+            energy_grid = grid
+        else:
+            energy_grid = nuc.energy[nucT]
+
+        for i, mt_set in enumerate(mts):
+            # Get the reaction xs data from the nuclide
+            funcs = []
+            op = ops[i]
+            for mt in mt_set:
+                if mt == 2:
+                    if sab_name:
+                        # Then we need to do a piece-wise function of
+                        # The S(a,b) and non-thermal data
+                        sab_sum = openmc.data.Sum(sab_funcs)
+                        pw_funcs = openmc.data.Regions1D(
+                            [sab_sum, nuc[mt].xs[nucT]],
+                            [sab_Emax])
+                        funcs.append(pw_funcs)
+                    else:
+                        funcs.append(nuc[mt].xs[nucT])
+                elif mt == 5 and mt in nuc:
+                    # Only consider the (n,misc) products if neutrons are
+                    # included in the outgoing channel since (n,misc) is only
+                    # explicitly needed for scatter cross sections.
```

Comment:
In every case I know of, MT=5 has a neutron in the exit channel. I checked all nuclides in ENDF/B-VII.1, B-VIII.0 beta, and JEFF 3.2 and there are no cases where MT=5 doesn't have a neutron. So, you probably don't need a special block for MT=5.